### ISSUE TITLE
Teleport Kube operator shortNames creates a conflict between predefined Kubernetes resources (Role)

### ISSUE THREAD

**Issue by WilliamLoy** [gravitational/teleport#17587]
Expected behavior:
Kubernetes should prioritize its own built-in api-resources. 

Current behavior:
(Teleport Operator)[https://github.com/gravitational/teleport/blob/master/examples/chart/teleport-clus[…]s/teleport-operator/templates/resources.teleport.dev_roles.yaml](https://github.com/gravitational/teleport/blob/master/examples/chart/teleport-cluster/charts/teleport-operator/templates/resources.teleport.dev_roles.yaml#L9-L15)) creates the teleportroles custom resource with shortNames of role and roles … that overrides the default behavior of Kubernetes and a kubectl get role will NOT return the kubernetes roles anymore but teleportroles.

Bug details:
- Teleport version: 10.2
- Recreation steps

1. Setup Teleport Kube Operator
2. observe `kubectl get roles` overriding default kube behavior as well as producing inconsistent output.

- Debug logs

**Comment by hugoShaka** [comment]
This is a really interesting issue and I believe kubectl's behaviour is not right.

## Reproduction

### Setup

```shell
# create anew cluster and save apis

k3d cluster create
kubectl get --raw /apis > 00-pre-install-apis.json
kubectl get --raw /api > 00-pre-install-api.json

# install our CRDs
helm install hugo-teleport-cluster teleport/teleport-cluster \
                          --set clusterName=hugo-teleport-cluster.teleport-cluster.svc.cluster.local \
                          --create-namespace \
                          --set operator.enabled=true \
                          --namespace=teleport-cluster \
                          --version 10.2.1

kubectl get --raw /apis > 01-post-install-apis.json
kubectl get --raw /api > 01-post-install-api.json
```

### Before getting a CR

Get role works as expected, it prioritizes names above shortnames.

```shell
kubectl get role -n teleport-cluster
NAME                        CREATED AT
hugo-teleport-cluster-psp   2022-10-20T19:17:35Z

kubectl get roles -n teleport-cluster
NAME                        CREATED AT
hugo-teleport-cluster-psp   2022-10-20T19:17:35Z
```

### After getting a CR

But then we get only once the teleport resource and it changes definetly the `kubectl get role` behaviour

```shell
kubectl get teleportrole -n teleport-cluster
No resources found in teleport-cluster namespace.

shaka@gazpacho ~> kubectl get role -n teleport-cluster
No resources found in teleport-cluster namespace.

shaka@gazpacho ~> kubectl get roles -n teleport-cluster
NAME                        CREATED AT
hugo-teleport-cluster-psp   2022-10-20T19:17:35Z

kubectl get --raw /api > 02-after-invoke-api.json
kubectl get --raw /apis > 02-after-invoke-apis.json
```

Note: this behaviour is not flaky, you can retry each step as many times as you want you will always get the same output.

I compared `/api` and `/apis` answers, there is no difference between before and after getting a CR. The only difference happens during CRD installation, which makes sense:
```diff
diff 00-pre-install-apis.json.linted 01-post-install-apis.json.linted
331a332,348
>       "name": "resources.teleport.dev",
>       "versions": [
>         {
>           "groupVersion": "resources.teleport.dev/v5",
>           "version": "v5"
>         },
>         {
>           "groupVersion": "resources.teleport.dev/v2",
>           "version": "v2"
>         }
>       ],
>       "preferredVersion": {
>         "groupVersion": "resources.teleport.dev/v5",
>         "version": "v5"
>       }
>     },
>     {
```

## Next steps

I suspect the `kubectl`/go-client is caching stuff on disk, but instead of caching the `/apis/X` answers it caches the result of the RESTMapping resolution. This causes subsequent resolutions to bypass the priority logic added by https://github.com/kubernetes/kubernetes/pull/58861. (cache is one of [the 2 hard things](http://thecodelesscode.com/case/220) after all :upside_down_face: )

We might want to open another issue because [the previous one](https://github.com/kubernetes/kubernetes/issues/59812) was closed.

In the meantime we have to avoid shortname conflicts. I will implement a blocklist in `crdgen` so we can avoid problematic shortnames while continuing to provide useful shortnames like `oidcconnector` instead of `teleportoidcconnector`.
